src/container/streamAnalyticsManager.ts

```diff
@@ -29,40 +29,35 @@ export class StreamAnalyticsManager {
         return job;
     }
 
-    public getImageName(): string {
-        return this.STREAM_ANALYTICS_IMAGE;
-    }
-
     public async getJobInfo(streamingJob: StreamAnalyticsPicItem): Promise<object> {
-        const subscriptionId: string = streamingJob.azureSubscription.subscription.id;
-        const id: string = streamingJob.job.id;
-        const resourceGroup: string = id.slice(id.toLowerCase().search("resourcegroups/") + "resourcegroups/".length, id.toLowerCase().search("/providers/"));
-        const jobName: string = streamingJob.job.name;
-        const publishJobUrl: string = `https://management.azure.com${subscriptionId}/resourceGroups/${resourceGroup}` +
-                                    `/providers/Microsoft.StreamAnalytics/streamingjobs/${jobName}/publishedgepackage?api-version=2017-04-01-preview`;
-
-        const { aadAccessToken } = await this.acquireAadToken(streamingJob.azureSubscription.session);
-
-        const publishResponse = await request.post(publishJobUrl, {
-            auth: {
-                bearer: aadAccessToken,
-            },
-            resolveWithFullResponse: true,
-        });
-
-        const operationResultUrl = publishResponse.headers.location;
-
-        const jobInfoResult = await request.get(operationResultUrl, {
-            auth: {
-                bearer: aadAccessToken,
-            },
-        });
-
         try {
+            const id: string = streamingJob.job.id;
+            const publishJobUrl: string = `https://management.azure.com${id}/publishedgepackage?api-version=2017-04-01-preview`;
+
+            const { aadAccessToken } = await Utility.acquireAadToken(streamingJob.azureSubscription.session);
+
+            const publishResponse = await request.post(publishJobUrl, {
+                auth: {
+                    bearer: aadAccessToken,
+                },
+                resolveWithFullResponse: true,
+            });
+
+            const operationResultUrl = publishResponse.headers.location;
+
+            await this.sleep(1000);
```

Comment:
Why do we need to sleep here?

Reply:
Because the rest api is asynchronous, we need to wait in case of error

Reply:
Is there any way we can know whether the operation succeeds instead of hard-coding a sleep time?